### ISSUE TITLE
PointLight limit?

### ISSUE THREAD

**Issue by miccTronic** [helix-toolkit/helix-toolkit#1859]
Hello,

HelixToolkit is truly an amazing piece of work and I'm using the SharpDX version already in several of my projects. However, currently I am trying to light a scene (a game map) with point lights, and I get no more lighting after the 8th light. So, while the first 8 lights are working perfectly, any lights I add after the 8th are simply dark. I am adding the lights using

`SceneNodeGroupModel3D.AddNode(new PointLightNode() { Range = a, Position = b, Color = c });`

Is this a known limitation, or am I doing something wrong?
Thanks!

**Comment by holance** [comment]
Yes, we are currently limited number of lights to 8 for performance purpose.

**Comment by miccTronic** [comment]
Thanks holance for the quick info. Are there any plans to increase this limit any time soon, or do you think it would be possible for me to change this limitation in source and compile a custom version?

I could probably dynamically turn on and off some lights outside the player's range, but it still makes it hard to illuminate a level with this limit in place (and I didn't test but assume these include directional lights too).

**Comment by holance** [comment]
You can change both here: 
https://github.com/helix-toolkit/helix-toolkit/blob/develop/Source/HelixToolkit.Native.ShaderBuilder/Common/DataStructs.hlsl#L7

and

https://github.com/helix-toolkit/helix-toolkit/blob/b7807188542288ab1c0efb11ba8bb113bafb47a6/Source/HelixToolkit.SharpDX.Shared/Interface/Constants.cs#L52

to increase number of light support.

**Comment by miccTronic** [comment]
Thanks for the hint with the shader file, I wouldn't have found that. So I'll try recompiling with an increased number.

Right now I already implemented dynamic on/off based on distance, but you still notive some lights switching because there simply are a number of them. If the increase is not enough, I'll try that other hint with the regions.
